packages/app-router/src/hooks.tsx

```diff
@@ -0,0 +1,620 @@
+   */
+  async function push(path: string, passProps = {}, options?: Options) {
+    const matchedRoute = route.routes.find(it => {
+      return match(it.path)(path);
+    });
+
+    if (!matchedRoute) return;
+
+    if (matchedRoute.stackId) {
+      Navigation.mergeOptions(componentId, {
+        bottomTabs: {
+          currentTabId: matchedRoute.stackId,
+        },
+      });
+    }
+
+    return Navigation.push(matchedRoute.stackId ?? componentId, {
+      component: {
+        name: matchedRoute.name,
+        passProps: {
+          ...passProps,
+          APP_ROUTER_URL: path, // Inject the URL path as a special prop
+        },
+        options,
+      },
+    });
+  }
+
+  /**
+   * Pop the current screen off the navigation stack, returning to the previous screen.
+   *
+   * @param {Options} [mergeOptions] - Optional navigation options for customizing the transition.
+   */
+  function pop(mergeOptions?: Options) {
+    return Navigation.pop(componentId, mergeOptions);
+  }
+
+  /**
+   * Pop all screens off the stack and return to the root screen.
+   *
+   * @param {Options} [mergeOptions] - Optional navigation options for customizing the transition.
+   */
+  function popToRoot(mergeOptions?: Options) {
+    return Navigation.popToRoot(componentId, mergeOptions);
+  }
+
+  /**
+   * Pop to a specific screen in the stack.
+   *
+   * @param {Options} [mergeOptions] - Optional navigation options for customizing the transition.
+   */
+  function popTo(mergeOptions?: Options) {
+    return Navigation.popTo(componentId, mergeOptions);
+  }
+
+  /**
+   * Set a new root for the stack, replacing all existing screens.
+   *
+   * @param {Layout | Layout[]} layout - The new layout or stack of layouts to set as the root.
+   */
+  function setStackRoot(layout: Layout | Layout[]) {
+    return Navigation.setStackRoot(componentId, layout);
+  }
+
+  /**
+   * Displays a modal with the specified component and data, and returns a promise
+   * that resolves when the modal is dismissed.
+   *
+   * @template T - The type of data passed to the modal.
+   * @template U - The type of data returned when the modal is resolved.
+   *
+   * @param Component - The React component to render inside the modal.
+   * @param data - The data to pass to the modal component.
+   * @param options - Additional options for displaying the modal.
+   *
+   * @returns A promise that resolves with the data returned from the modal when it's dismissed.
+   *
+   * @example
+   * ```typescript
+   * const result = await showModal<MyComponentProps, ResultType>(MyComponent, { prop1: 'value' }, { modalOptions: true });
+   * console.log('Modal result:', result);
+   * ```
+   */
+  async function showModal<T, U>(
+    Component: React.ComponentType,
+    data: T,
+    options: Options = {},
+  ): Promise<U> {
+    // Generate a unique name for the modal component.
+    const name = `Modal_${idCounter}`;
+    idCounter++;
```

Comment:
it might also be useful to optionally pass this `nane`  as we do for `route`

Reply:
I agree with this - but it needs more thought because we won't know for sure if the "screen" has been registered or not.